lib/bolt/util.rb

```diff
@@ -169,12 +169,13 @@ def postwalk_vals(data, skip_top = false, &block)
       # references to the same object and prevents endless recursion.
       # Credit to Jan Molic via https://github.com/rubyworks/facets/blob/master/LICENSE.txt
       def deep_clone(obj, cloned = {})
-        cloned[obj.object_id] if cloned.include?(obj.object_id)
+        return cloned[obj.object_id] if cloned.include?(obj.object_id)
 
         begin
           cl = obj.clone
-        rescue TypeError
-          # unclonable (TrueClass, Fixnum, ...)
+        rescue TypeError, Java::JavaLang::CloneNotSupportedException
```

Comment:
This will fail with an undefined constant in non-JRuby if an error other than TypeError is raised.